drivers/spi/spi_ll_stm32.c

```diff
@@ -0,0 +1,457 @@
+/*
+ * Copyright (c) 2016 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_SPI_LEVEL
+#include <logging/sys_log.h>
+
+#include <misc/util.h>
+#include <kernel.h>
+#include <board.h>
+#include <errno.h>
+#include <spi.h>
+
+#include <clock_control/stm32_clock_control.h>
+#include <clock_control.h>
+
+#include <drivers/spi/spi_ll_stm32.h>
+#include <spi_ll_stm32.h>
+
+#define CONFIG_CFG(cfg)						\
+((const struct spi_stm32_config * const)(cfg)->dev->config->config_info)
+
+#define CONFIG_DATA(cfg)					\
+((struct spi_stm32_data * const)(cfg)->dev->driver_data)
+
+#ifdef CONFIG_SPI_STM32_INTERRUPT
+static void spi_stm32_transmit(SPI_TypeDef *spi, struct spi_stm32_data *data);
+static void spi_stm32_receive(SPI_TypeDef *spi, struct spi_stm32_data *data);
+
+static void spi_stm32_isr(void *arg)
+{
+	struct device * const dev = (struct device *) arg;
+	const struct spi_stm32_config *cfg = dev->config->config_info;
+	struct spi_stm32_data *data = dev->driver_data;
+	SPI_TypeDef *spi = cfg->spi;
+
+	if (LL_SPI_IsActiveFlag_TXE(spi) && spi_context_tx_on(&data->ctx)) {
+		spi_stm32_transmit(spi, data);
+	}
+
+	if (LL_SPI_IsActiveFlag_RXNE(spi)) {
+		if (spi_context_rx_on(&data->ctx)) {
+			spi_stm32_receive(spi, data);
+		} else {
+			LL_SPI_DisableIT_RXNE(spi);
```

Comment:
TX must continue until RX bytes are exhausted, this is why we ignore RX irqs when only TX remains.